notebooks/document-chunking/with-index-pipelines.ipynb

```diff
@@ -0,0 +1,561 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "id": "_OAahfg-tqrf",
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "_OAahfg-tqrf",
+    "outputId": "d8f81ba4-cdc9-4e30-edf7-6d5bb16920eb"
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "ObjectApiResponse({'acknowledged': True, 'shards_acknowledged': True, 'index': 'chunker'})"
+      ]
+     },
+     "execution_count": 17,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "\n",
+    "INDEX_NAME = \"chunk_passages_example\"\n",
+    "\n",
+    "# Setup the index\n",
+    "client.indices.create( \n",
+    "  index=INDEX_NAME, \n",
+    "  settings={\n",
+    "    \"index\": {\n",
+    "      \"number_of_replicas\": \"1\",\n",
+    "      \"number_of_shards\": \"1\",\n",
+    "      \"default_pipeline\": \"chunk_text_to_passages\"\n",
+    "    }\n",
+    "  },\n",
+    "  mappings={\n",
+    "    \"dynamic\": \"true\",\n",
+    "    \"properties\": {\n",
+    "      \"passages\": {\n",
+    "        \"type\": \"nested\",\n",
+    "        \"properties\": {\n",
+    "          \"vector\": {\n",
+    "            \"properties\": {\n",
+    "              \"predicted_value\": {\n",
+    "                \"type\": \"dense_vector\",\n",
+    "                \"index\": True,\n",
+    "                \"dims\": 384,\n",
+    "                \"similarity\": \"dot_product\"\n",
+    "              }\n",
+    "            }\n",
+    "          }\n",
+    "        }\n",
+    "      }\n",
+    "    }\n",
+    "  }\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "075f5eb6",
+   "metadata": {
+    "id": "075f5eb6"
+   },
+   "source": [
+    "### Add some Documents\n",
+    "\n",
+    "Now we can add documents with large amounts of text in body_content and automatically have them chunked, and each chunk text embedded into vectors by the model:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "id": "008d723e",
+   "metadata": {
+    "id": "008d723e"
+   },
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "from urllib.request import urlopen\n",
+    "from elasticsearch import helpers\n",
+    "\n",
+    "url = \"https://raw.githubusercontent.com/elastic/elasticsearch-labs/main/datasets/workplace-documents.json\"\n",
+    "docs = json.loads(urlopen(url).read())\n",
+    "\n",
+    "operations = [\n",
+    "    {\n",
+    "      \"_index\": INDEX_NAME,\n",
+    "      \"_id\": i,\n",
+    "      \"text\": doc[\"content\"],\n",
+    "      \"name\": doc[\"name\"]\n",
+    "    } for i, doc in enumerate(docs)\n",
+    "]\n",
+    "\n",
+    "# Add the documents to the index directly\n",
+    "response = helpers.bulk(\n",
+    "  client,\n",
+    "  operations,\n",
+    ")"
```

Comment:
I think it needs `time.sleep` after helpers.bulk operation to pass the`nbtest`.

Reply:
added a refresh flag so docs are immediately available to the search query.